beacon_node/rest_api/tests/test.rs

```diff
@@ -303,8 +303,11 @@ fn check_duties<T: BeaconChainTypes>(
 
             if !duty.block_proposal_slots.is_empty() {
                 for slot in &duty.block_proposal_slots {
+                    state
+                        .update_proposer_indices_cache(*slot, &spec)
+                        .expect("should update proposer indices cache");
```

Comment:
We need to constantly be updating this in the for loop in case we transition in another epoch